src/bentoml/_internal/frameworks/diffusers.py

```diff
@@ -0,0 +1,355 @@
+class DiffusersOptions(PartialKwargsModelOptions):
+    """Options for the diffusers model."""
+
+    pipeline_class: type[diffusers.pipelines.DiffusionPipeline] | None = None
+    scheduler_class: type[diffusers.SchedulerMixin] | None = None
+    torch_dtype: str | torch.dtype | None = None
+    custom_pipeline: str | None = None
+    enable_xformers: bool | None = None
+
+
+def get(tag_like: str | Tag) -> bentoml.Model:
+    """
+    Get the BentoML model with the given tag.
+
+    Args:
+        tag_like: The tag of the model to retrieve from the model store.
+
+    Returns:
+        :obj:`~bentoml.Model`: A BentoML :obj:`~bentoml.Model` with the matching tag.
+
+    Example:
+
+    .. code-block:: python
+
+       import bentoml
+       # target model must be from the BentoML model store
+       model = bentoml.diffusers.get("my_stable_diffusion_model")
+    """
+    model = bentoml.models.get(tag_like)
+    if model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {model.tag} was saved with module {model.info.module}, not loading with {MODULE_NAME}."
+        )
+    return model
+
+
+def load_model(
+    bento_model: str | Tag | bentoml.Model,
+    pipeline_class: type[
+        diffusers.pipelines.DiffusionPipeline
+    ] = diffusers.StableDiffusionPipeline,
+    device_map: str | dict[str, int | str | torch.device] = "auto",
+    custom_pipeline: str | None = None,
+    scheduler_class: type[diffusers.SchedulerMixin] | None = None,
+    torch_dtype: str | torch.dtype | None = None,
+    low_cpu_mem_usage: bool | None = None,
+    enable_xformers: bool = False,
+) -> diffusers.DiffusionPipeline:
+    """
+    Load a Diffusion model and convert it to diffusers `Pipeline <https://huggingface.co/docs/diffusers/api/pipelines/overview>`_
+    with the given tag from the local BentoML model store.
+
+    Args:
+        bento_model: Either the tag of the model to get from the store, or a BentoML
+            ``~bentoml.Model`` instance to load the model from.
+
+    Returns:
+        The Diffusion model loaded as diffusers pipeline from the BentoML model store.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+        pipeline = bentoml.diffusers.load_model('my_diffusers_model:latest')
+        pipeline(prompt)
+    """  # noqa
+    if not isinstance(bento_model, bentoml.Model):
+        bento_model = get(bento_model)
+
+    if bento_model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {bento_model.tag} was saved with module {bento_model.info.module}, not loading with {MODULE_NAME}."
+        )
+
+    if pipeline_class is None:
+        pipeline_class = diffusers.StableDiffusionPipeline
+
+    diffusion_model_dir = bento_model.path_of(DIFFUSION_MODEL_FOLDER)
+
+    if low_cpu_mem_usage is None:
+        if is_torch_version(">=", "1.9.0"):
+            low_cpu_mem_usage = True
+        else:
+            low_cpu_mem_usage = False
```

Comment:
any specific reason why we are doing this?

Reply:
What if `low_cpu_mem_usage` is passed in as `True` but torch version < 1.9.0

Reply:
This code will automatically pick a suitable value for `low_cpu_mem_usage`.

If user pass `low_cpu_mem_usage=True` explicitly but torch version < 1.9.0, `diffuesrs` will raise an error. I think it's better to let `diffusers` to raise the error instead of silently converting `low_cpu_mem_usage` to `False` by ourselves. It will give user a clear warning/error and we don't override the future behavior of diffusers if they decide to change this part.